packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php

```diff
@@ -20,30 +23,60 @@ public function __construct(
         private readonly ReflectionResolver $reflectionResolver,
         private readonly TypeComparator $typeComparator,
         private readonly StaticTypeMapper $staticTypeMapper,
+        private readonly PrivatesAccessor $privatesAccessor,
     ) {
     }
 
-    public function hasParentClassMethod(ClassMethod $classMethod): bool
+    public function hasParentClassMethod(ClassMethod $classMethod): ?bool
```

Comment:
this method now return `true/false` when we can make a final decision or `null` when not classes involved are known and we can't judge